### DISCUSSION TITLE
update Pocketbase inputs ReactJs

### DISCUSSION THREAD

**Discussion by richzx** [pocketbase/pocketbase#5385] [Q&A]
how can I update only a few inputs and have the other inputs keep the information already entered?

![image](https://github.com/user-attachments/assets/cfedd9bd-2ab7-49eb-ae83-d524e27f3f30)
![image](https://github.com/user-attachments/assets/6172bd50-1e35-4e0e-b8e0-2337b693ca8b)
![image](https://github.com/user-attachments/assets/9ff97082-6dde-4962-9b5c-07b5fde10982)

**Comment by edions** [comment]
If I understand correctly, you want to filter out the data from being updated if the input value is empty. If so you can use the `Object.entries()` and `reduce()` functions to filter out any null or undefined values.

Example:
```jsx
export async function updateClient(id, value, numberDoc, nameClient, laNameClient,
directionClient, phoneClient, emailClient) {
    const inputData = {
        "tipe_document": value,
        "numberdoc_client": numberDoc,
        "name_client": nameClient,
        "lastname_client": laNameClient,
        "direction_client": directionClient,
        "phone_client": phoneClient,
        "email_client": emailClient
    };

    // Filter out data that are null or undefined
    const data = Object.entries(inputData).reduce((acc, [key, val]) => {
        if (val !== undefined && val !== null) {
            acc[key] = val;
        }
        return acc;
    }, {});

    await dpP.collection("addClient").update(id, data);
}
```
